frontend/src/utils/lessonContext.js

```diff
@@ -0,0 +1,35 @@
+import { createContext, useState, useEffect, useReducer } from "react";
+import { useParams } from "react-router-dom";
+
+import { startLesson } from "../services/lessonService";
+
+const LessonContext = createContext(null);
+
+const LessonContextProvider = ({ children }) => {
+  const [lesson, setLesson] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [lessonId, setLessonId] = useState();
+
+  useEffect(() => {
+    const fetchData = () => {
+      startLesson(lessonId)
+        .then(({ data }) => {
+          setLesson(data);
+          setLoading(false);
+        })
+        .catch(() => {
+          setLoading(false);
+        })
+        .finally(() => {});
+    };
+    fetchData();
```

Comment:
```suggestion
      startLesson(lessonId)
        .then(({ data }) => {
          setLesson(data);
          setLoading(false);
        })
        .catch(() => {
          setLoading(false);
        })
```

1. no need to wrap it in a function, this is not async await, it is a promise. So ok not to wrap. 
2. Its fine to remove finally if nothing is in there